circuit/types/group/src/lib.rs

```diff
@@ -57,12 +57,43 @@ impl<E: Environment> Inject for Group<E> {
     /// For safety, the resulting point is always enforced to be on the curve with constraints.
     /// regardless of whether the y-coordinate was recovered.
     fn new(mode: Mode, group: Self::Primitive) -> Self {
-        // Initialize the x- and y-coordinate field elements.
-        let (x, y) = group.to_xy_coordinate();
-        let x = Field::new(mode, x);
-        let y = Field::new(mode, y);
+        // Inject the point.
+        let point = {
+            // Initialize the (x, y) coordinates of the point as field elements.
+            let (x, y) = group.to_xy_coordinate();
+            // Inject the `(x, y)` coordinates as field elements.
+            Self { x: Field::new(mode, x), y: Field::new(mode, y) }
+        };
+
+        // Inject `(point / COFACTOR)` as a witness.
+        let point_inv: Self = witness!(|point| point.div_by_cofactor());
```

Comment:
We don't need to separately allocate both `point_inv` and `point`; we just allocate `point_inv`, check that the result is on the curve, and then output `point_inv.mul_by_cofactor()`. This avoids extra allocations and also the equality check.